Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMDecryptorGStreamer.cpp

```diff
@@ -105,11 +132,13 @@ static gboolean webKitMediaOpenCDMDecryptorHandleKeyResponse(WebKitMediaCommonEn
 
     GST_DEBUG_OBJECT(self, "handling session %s for event %u (ours %u)", session.get(), protectionEvent, priv->m_protectionEvent);
     if (priv->m_protectionEvent == protectionEvent) {
-        if (priv->m_session != session.get()) {
+        if (priv->m_openCdm == nullptr) {
             priv->m_session = session.get();
             priv->m_openCdm = std::make_unique<media::OpenCdm>(priv->m_session.utf8().data());
-            GST_DEBUG_OBJECT(self, "selected session %s", priv->m_session.utf8().data());
             returnValue = true;
+        }
+        else if (priv->m_session != session.get()) {
+            GST_DEBUG_OBJECT(self, "Different session send to:  %s - %s", priv->m_session.utf8().data(), session.get());
```

Comment:
Would write this has "selected a new session %s, old one was %s" to make it easier to understand

Reply:
It is not selecting a new session. It is reporting that a new session is now calling this method, compared to the one previously set. I think this should not happen. Might put an assert here.
I think changing the logic that if a new session is send, we might take it up as being the active session requires a more intrusive approach. Than we need to think about locking as the current session might be in use with a decrypt and we just replace it by a new one. The sessions should become than reference counted and it will, potentially, bring more locking consequences (and overhead) for something that I think, is form a design impossible.

Reply:
Regarding the semantics of session selection, the message should be formatted like @charlie-ht says.